R/nm-tables.R

```diff
@@ -84,17 +84,18 @@ nm_tables <- function(
 #' @export
 nm_table_files <- function(.mod, .check_exists = TRUE) {
   .p <- get_model_path(.mod)
-  .l <- parse_ctl_to_list(.p)
+  .l <- nmrec::read_ctl(.p)
   out_dir <- get_output_dir(.mod, .check_exists = .check_exists)
 
   # get file names from table statements and construct paths
-  .f <- .l[grep("^TAB", names(.l))] %>%
-    map_chr(~paste(.x, collapse = " ")) %>%
-    str_extract("\\bFILE\\s*=\\s*([^ ]+)") %>%
-    str_replace("\\bFILE\\s*=\\s*", "") %>%
-    str_replace("^\\.\\/", "") %>%
+  .f <- nmrec::select_records(.l, "table") %>%
+    purrr::map(function(.x) nmrec::get_record_option(.x, "file")$value) %>%
+    unlist() %>% str_replace("^\\.\\/", "") %>%
     file.path(out_dir, .)
 
+  # Coerce to the same type of quote
+  .f <- sub("^'(.*)'$", "\\1", .f) %>% sub('^"(.*)"$', "\\1", .)
```

Comment:
Sorry, I realize this was my suggestion, but this code will incorrectly strips a set of inner double quotes, while the expected quoting rules would be that only the outer layer is stripped.

I've confirmed that this is how NONMEM treats the quoting:

```
$ grep TAB 1.ctl
$TABLE NUM IPRED NPDE CWRES NOPRINT ONEHEADER FILE='"1.tab"'

$ ls --quoting-style=literal 1/*tab*
1/"1.tab"
```

How about something like this instead?

```R
quoted <- grepl("^'.*'$", .f) | grepl('^".*"$', .f)
if (any(quoted)) {
  .f[quoted] <- stringr::str_sub(.f[quoted], 2, -2)
}
```

You could of course pull that into something like an `unquote_filename`, if you find that clearer.

Reply:
@kyleam in your first code snippet, is that the expected behavior, or an example of incorrect filename specifications?

Trying to get a better understanding of what is accepted. Is it A) no quotes or B) two sets of quotes (double for the inner)?

Reply:
> in your first code snippet, is that the expected behavior, or an example of incorrect filename specifications?

It's verification that NONMEM will only strip the outer layer of quotes (as expected for shell quoting).   Line 97 in the current code will incorrectly strip the inner layer of quotes.

> Trying to get a better understanding of what is accepted. Is it A) no quotes or B) two sets of quotes (double for the inner)?

Sorry, I don't follow.  Accepted by what?

Go into a bash shell and do this:

```
$ touch '"foo"'
```

That creates a file `"foo"` (double quotes are part of the actual name).  Likewise, `touch "'foo'"` would create a file `'foo'`.  NONMEM follows that expected behavior of only removing a single layer of quoting.

For an R return value, we want to return the name of the actual file, without any quoting.  The current code on line 97 doesn't because it will strip `'"foo"'` to `foo`, when the actual file name is `"foo"`.

Reply:
Thanks for the added clarity. I went with creating a new function like you suggested, though I opted for more of an integration test (as opposed to testing `unquote_filename` specifically)